### DISCUSSION TITLE
Using more than one HMS inverter

### DISCUSSION THREAD

**Discussion by KFU141** [DennisOSRM/hms-mqtt-publisher#114] [Q&A] [answered]
Hi,
is it possible to configure the config.toml in a way that more than one inverter can run in parallel? Or is it necessary to e.g. run a second docker instance?

At least what I see a second docker instance is not a solution because it will lead to simple MQTT messages with the same topic "hms800wt2".

What about using an additional variable in the config.toml with which the topic can be defined together with an additional inverter IP?

**Comment by KFU141** [comment, marked as answer]
Hi,

I found a workaround changing the topics in the simple_mqtt.rs and creating an additional docker image. Now I am running two containers with different MQTT topics and IPs for both the inverters.

Will create an issue request for a build in solution.

Cheers

**Comment by dk8pn** [comment]
Hello,

i did what you proposed here. It basically works fine. But i observed one issue:
In openHAB two sets of Homeassistant sensors (18 sensors each) have been detected as new things. Working with one inverter, this thing is always "Online". Working with two inverters, both things switch between "Online" and "Unknown". Data seems not to be affected. Any idea for what reason this happens?

**Comment by joha38** [comment]
Hello, I have 2 hms-1000w-2t

how to create a new docker image ?

Thanks

**Reply by dk8pn** [reply]
I am a Windows10 user and use "WSL" (Windows Subsysten for Linux). A simply made a copy of my first folder "hms-mqtt-publisher", generated from the docker image, to a new folder "hms-mqtt-publisher-2". Then i edited the file acc. to @KFU141 in order to have unique MQTT elements for each inverter. Finally i start a 2nd WSL instance and both inverters are running.

**Reply by joha38** [reply]
Thanks for the answer, but I use docker on debian and I would like to have 2 dockers

**Comment by Diemo79** [comment]
Hi, I hope you can help me.

What do I need to change, and where, so that I can distinguish between multiple inverters in ioBroker?

For example, as shown in the picture, where can I change the “0” to a “1”, or change the name “hms800wt2”?

![mqtt-structur](https://github.com/user-attachments/assets/b320f32a-5d1e-4a6d-9092-2fe8612bd622)

Thanks

**Reply by dk8pn** [reply]
Please read the few posts above. They describe what needs to be done.
